lang/java/avro/src/main/java/org/apache/avro/Conversions.java

```diff
@@ -0,0 +1,109 @@
+package org.apache.avro;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericFixed;
+
+public class Conversions {
+
+  public static class UUIDConversion extends Conversion<UUID> {
+    @Override
+    public Class<UUID> getConvertedType() {
+      return UUID.class;
+    }
+
+    @Override
+    public Schema getRecommendedSchema() {
+      return LogicalTypes.uuid().addToSchema(Schema.create(Schema.Type.STRING));
+    }
+
+    @Override
+    public String getLogicalTypeName() {
+      return "uuid";
+    }
+
+    @Override
+    public UUID fromCharSequence(CharSequence value, Schema schema, LogicalType type) {
+      return UUID.fromString(value.toString());
+    }
+
+    @Override
+    public CharSequence toCharSequence(UUID value, Schema schema, LogicalType type) {
+      return value.toString();
+    }
+  }
+
+  public static class DecimalConversion extends Conversion<BigDecimal> {
+    @Override
+    public Class<BigDecimal> getConvertedType() {
+      return BigDecimal.class;
+    }
+
+    @Override
+    public Schema getRecommendedSchema() {
+      throw new UnsupportedOperationException(
+          "No recommended schema for decimal (scale is required)");
+    }
+
+    @Override
+    public String getLogicalTypeName() {
+      return "decimal";
+    }
+
+    @Override
+    public BigDecimal fromBytes(ByteBuffer value, Schema schema, LogicalType type) {
+      int scale = ((LogicalTypes.Decimal) type).getScale();
+      byte[] bytes;
+      if (value.hasArray()) {
+        bytes = value.array();
```

Comment:
TODO: this assumes the backing array is valid from 0 to its length, which isn't necessarily the case. Need to use an offset/length constructor for the unscaled BigInteger value.